### PR TITLE
[feat]: [PLG-3024]: added isCannyUsernameAbbreviationEnabled into NG API response

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes

**API**: @GET https://localhost:8181/ng/api/accounts/kmpySmUISimoRrJL6NL73w?routingId=kmpySmUISimoRrJL6NL73w

**response**: 
{
    "status": "SUCCESS",
    "data": {
        "identifier": "kmpySmUISimoRrJL6NL73w",
        "name": "Harness",
        "companyName": "Harness",
        "cluster": "",
        "defaultExperience": "CG",
        "authenticationMechanism": "USER_PASSWORD",
        "createdAt": 1684757371309,
        "ringName": "ring3",
        "sessionTimeoutInMinutes": 1440,
        "publicAccessEnabled": false,
        "twoFactorAdminEnforced": false,
        "nextGenEnabled": true,
        "productLed": false,
        "harnessSupportAccessAllowed": false,
        "crossGenerationAccessEnabled": false,
        "cannyUsernameAbbreviationEnabled": true
    },
    "metaData": null,
    "correlationId": "dbf23ab7-d9a5-4f71-b2da-568ca197ea8b"
}

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55406)

### CODE REVIEW

**Comment by kapil-harness** [comment]
trigger uts

**Review by mohitg0795 — APPROVED** [review]

**Review by Xin-Harness — APPROVED** [review]

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java [hunk 1]
```diff
@@ -53,6 +53,9 @@ public class AccountDTO {
   @Schema(description = "Specifies weather access to other generation is allowed or not")
   @VariableExpression(skipVariableExpression = true)
   boolean isCrossGenerationAccessEnabled;
+  @Schema(description = "Specifies weather CannyUsernameAbbreviation is enabled for this account")
```
Comment (neelam-harness) [inline 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java:56]: typo: `weather` -> `whether`

**Review by neelam-harness — CHANGES_REQUESTED** [review]

File: 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java (hunk 1 above)
Comment (kapil-harness) [reply 960-ng-core-beans/src/main/java/io/harness/ng/core/dto/AccountDTO.java]: done

**Review by neelam-harness — APPROVED** [review]

**Review by mohitg0795 — APPROVED** [review]

**Comment by kapil-harness** [comment]
trigger review

**Comment by kapil-harness** [comment]
trigger review

**Pull request merged**